docs/website/docs/plus/getting-started/advanced_tutorial.md

```diff
@@ -0,0 +1,305 @@
+
+If you want to prevent dlt+ from creating implicit entities, you can set the `allow_undefined_entities` option to `false` in the project configuration by adding the following to your `dlt.yml` file:
+
+```yaml
+project:
+  allow_undefined_entities: false
+```
+
+If you try to run the pipeline "my_pipeline" now, you will receive an error that the dataset `my_pipeline_dataset` does not exist. To fix this, you can create the dataset explicitly in the `dlt.yml` file:
+
+```yaml
+datasets:
+  my_pipeline_dataset:
+    destination:
+        - duckdb
+```
+
+Note that the destination property is an array of destinations on which this dataset may be materialized. More on that in the next section.
+
+
+## Managing Datasets and Destinations
+
+If you add explicit dataset entries to the `dlt.yml` file, as we did above, you need to defined on which destinations the dataset may be materialized. This constraint will also be enforced when `allow_undefined_entities` is set to `true`. Change your dlt.yml file to the following:
+
+
+```yaml
+# your sources are the data sources you want to load from
+sources:
+  arrow:
+    type: sources.arrow.source
+
+# your destinations are the databases where your data will be saved
+destinations:
+  duckdb:
+    type: duckdb
+
+datasets:
+  my_pipeline_dataset:
+    destination:
+        - bigquery
+
+# your pipelines orchestrate data loading actions
+pipelines:
+  my_pipeline:
+    source: arrow
+    destination: duckdb
+    dataset_name: my_pipeline_dataset
+```
+
+If you try to run the pipeline "my_pipeline" now, you will receive an error that the dataset `my_pipeline_dataset` does not exist on the bigquery destination, you may allow this by adding `bigquery` to the allowed destinations of `my_pipeline_dataset`.
+
+```yaml
+my_pipeline_dataset:
+  destination:
+      - duckdb
+      - bigquery
+```
+
+:::tip
+If  you run dataset cli commands without providing a destination name, dlt+ will always select the first destination in the list by default. If you allow implicit entities, dlt+ can also discover datasets only defined on pipelines and will use the destination of that pipeline for instantiating the dataset. The same is true when retrieving datasets from the catalog in python, ore on that below.
+:::
+
+
+## Interacting with the project context in code
+
+You can also interact with your dlt+ project directly in code as opposed to running cli commands. Let's use the same starting point as above by running `dlt project init arrow duckdb`. Now add a new python file in the root of your project called `do_something.py` and add the following code:
```

Comment:
the above starting point was `dlt project init arrow duckdb --package my_dlt_project` so when I run `dlt project init arrow duckdb` in the same directory I see that new files are added to the current dir and not to `my_dlt_project`.

Reply:
so it's best either to instruct the user to clean the directory or to use the same init commands